### DISCUSSION TITLE
Typescript types for wrapper for new useStore(store) API.

### DISCUSSION THREAD

**Discussion by foxymiles** [pmndrs/zustand#1084] [Q&A] [answered]
The recommended method for using zustand with context since v4 is to use a vanilla store and the new `useStore` API that accepts a store as the first parameter and a plain react context and provider to contain the store. I'd like to write a wrapper function like this so that I don't have to call `useContext` first everywhere I use the store:

```typescript
function useMyStore(selector, equalityFn) {
  const store = useContext(StoreContext);
  const slice = useStore(store, selector, equalityFn);
  return slice;
}
```

But I'm not sure about the typescript types for `selector`, `equalityfn` and the return value. The types in `zustand/context` are quite complex. Has anybody figured this out already?

**Comment by dai-shi** [comment, marked as answer]
This has been asked before somewhere, so it could be a doc issue.
I think this should work.
```ts
function useMyStore<T>(selector: (state: MyState) => T, equalityFn: (a: T, b: T) => boolean) {
  const store = useContext(StoreContext);
  const slice = useStore(store, selector, equalityFn);
  return slice;
}
```

**Reply by foxymiles** [reply]
Thanks very much! I looked around but I wasn't able to find an example of this.
